lib/administration/resource/src/resource_administration.cpp

```diff
@@ -87,51 +84,48 @@ namespace irods::experimental::administration::NAMESPACE_IMPL
     {
         try {
-            const auto gql = fmt::format("select RESC_ID where RESC_NAME = '{}'", _name);
-            return {std::error_code{}, query_builder{}.build(_comm, gql).size() > 0};
+            return query_builder{}.build(_comm, fmt::format("select RESC_ID where RESC_NAME = '{}'", _name)).size() > 0;
+        }
+        catch (const irods::exception& e) {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+            THROW(e.code(), e.what());
+        }
+        catch (const std::exception& e) {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+            THROW(SYS_LIBRARY_ERROR, e.what());
         }
         catch (...) {
-            return {std::error_code{SYS_UNKNOWN_ERROR, std::generic_category()}, false};
+            constexpr char* msg = "An unknown error occurred while verifying existence of resource [{}]";
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+            THROW(SYS_UNKNOWN_ERROR, fmt::format(msg, _name));
         }
     }
 
-    auto resource_info(RxComm& _comm, const resource_name_type _name)
-        -> std::tuple<std::error_code, std::optional<class resource_info>>
+    auto resource_info(RxComm& _comm, const resource_name_type _name) -> std::optional<class resource_info>
     {
-        const auto to_resource_status = [](const std::string_view _status)
-        {
-            // clang-format off
-            if (_status == "up")   return resource_status::up;
-            if (_status == "down") return resource_status::down;
-            // clang-format on
-
-            return resource_status::unknown;
-        };
-
-        class resource_info info;
-        info.name_ = _name.data();
-
         try {
-            // clang-format off
-            const auto gql = fmt::format("select RESC_ID, RESC_TYPE_NAME, RESC_ZONE_NAME,"
-                                         "       RESC_LOC, RESC_VAULT_PATH, RESC_STATUS,"
-                                         "       RESC_CONTEXT, RESC_COMMENT, RESC_INFO,"
-                                         "       RESC_FREE_SPACE, RESC_FREE_SPACE_TIME,"
-                                         "       RESC_PARENT, RESC_CREATE_TIME, RESC_MODIFY_TIME "
-                                         "where RESC_NAME = '{}'", _name);
-            // clang-format on
+            const auto gql = fmt::format(
+                "select RESC_ID, RESC_TYPE_NAME, RESC_ZONE_NAME, "
+                "RESC_LOC, RESC_VAULT_PATH, RESC_STATUS, "
+                "RESC_CONTEXT, RESC_COMMENT, RESC_INFO, "
+                "RESC_FREE_SPACE, RESC_FREE_SPACE_TIME, "
+                "RESC_PARENT, RESC_CREATE_TIME, RESC_MODIFY_TIME "
+                "where RESC_NAME = '{}'",
+                _name);
 
             for (auto&& row : query_builder{}.build(_comm, gql)) {
+                class resource_info info;
+                info.name_ = std::string{_name};
```

Comment:
Why is this inside the loop now? I guess there will ever only be one result for this query, so this should be fine. Just curious

Reply:
The previous impl returned a resource_info object even when the resource didn't exist. This makes the logic clearer because you only get the resource_info object if the query found a match.

This also makes it clearer to see when std::nullopt is returned.

Reply:
Makes sense.